File: ThreadsApp/ThreadsApp/View/Chat/Cells/Select/SelectViewController.m
//
//  SelectViewController.m
//  ThreadsApp
//
//  Created by Tabriz Dzhavadov on 13/07/2017.
//  Copyright © 2017 Sequenia. All rights reserved.
//

#import "SelectViewController.h"

@interface SelectViewController () <UITableViewDelegate, UITableViewDataSource>

@property (strong, nonatomic) UITableView *tableView;

@end

@implementation SelectViewController

#pragma mark - Life Cycle

- (void) viewDidLoad {
    [super viewDidLoad];
    
    self.navigationItem.title = @"Design";
    
    [self configureTableView];
}

#pragma mark - Configure

- (void) configureTableView {
    self.tableView = [[UITableView alloc] initWithFrame: self.view.bounds];
    self.tableView.delegate = self;
    self.tableView.dataSource = self;
    self.tableView.rowHeight = UITableViewAutomaticDimension;
    self.tableView.estimatedRowHeight = 44.0f;
    [self.view addSubview:self.tableView];
}

#pragma mark - Table View

- (NSInteger) numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger) tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.selects.count;
}

- (UITableViewCell *) tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [UITableViewCell new];
    THRDesign design = [self.selects[indexPath.row] integerValue];
    cell.textLabel.text = [SelectCell designName: design];
    return cell;
}

- (void) tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath: indexPath animated: YES];
    if (self.delegate && [self.delegate respondsToSelector:@selector(selectCellDidSelect:)]) {
        THRDesign design = [self.selects[indexPath.row] integerValue];
        [self.delegate selectCellDidSelect: design];
    }
    [self dismissViewControllerAnimated: YES completion: nil];
}

@end

File: ThreadsApp/ThreadsAppSwift/View/Chat/SelectViewController.swift
//
//  SelectViewController.swift
//  THRAppSwift-Polarbear
//
//  Created by Brooma Service on 16/05/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

import Foundation
import UIKit


class SelectViewController : UIViewController, UITableViewDelegate, UITableViewDataSource {
    
    var selects : Array<THRDesign>?
    
    weak var delegate : SelectCellDelegate?
    
    var tableView : UITableView!
    
    //MARK: - Life Cycle
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.navigationItem.title = "Design"
        self.configureTableView()
    }
    
    //MARK: - Configure
    
    func configureTableView() {
        tableView = UITableView(frame: self.view.bounds)
        tableView.delegate = self
        tableView.dataSource = self
        tableView.rowHeight = UITableView.automaticDimension
        tableView.estimatedRowHeight = 44.0
        self.view.addSubview(tableView)
    }
    
    //MARK: - Table View
    
    func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.selects?.count ?? 0
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = UITableViewCell()
        let design = self.selects![indexPath.row]
        cell.textLabel?.text = SelectCell.designName(design: design)
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
        tableView.deselectRow(at: indexPath, animated: true)
        delegate?.selectCellDidSelect(design: self.selects![indexPath.row])
        self.dismiss(animated: true, completion: nil)
    }
}
